### PR TITLE
iOS notification to show sync status like android

### PR DESCRIPTION
**Pull request by neddoz**
What this PR does:
resolves [362](https://github.com/raedahgroup/dcrios/issues/362)
Shows the sync progress when the app is in background mode.
Image from iOS

### CODE REVIEW

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1069,11 +1075,12 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIconTestnet;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = U4FVESJ54S;
```
Comment (macsleven) [inline Decred Wallet.xcodeproj/project.pbxproj]: Please revert

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1152,11 +1159,12 @@
...
```
Comment (macsleven) [inline Decred Wallet.xcodeproj/project.pbxproj]: Please revert

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1296,11 +1304,12 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_IDENTITY = "iPhone Developer";
 				CODE_SIGN_STYLE = Automatic;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = U4FVESJ54S;
```
Comment (macsleven) [inline Decred Wallet.xcodeproj/project.pbxproj]: please revert

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -1322,11 +1331,12 @@
...
```
Comment (macsleven) [inline Decred Wallet.xcodeproj/project.pbxproj]: please revert

File: Podfile.lock [hunk 5]
```diff
@@ -39,4 +39,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 7da2d820dc0020e8a212c85fd7c1eae0ee0df943
 
-COCOAPODS: 1.7.5
+COCOAPODS: 1.7.3
```
Comment (macsleven) [inline Podfile.lock]: update pod

**Comment by macsleven** [comment]
The background task  time is minimal but it is an improvement from what we have.

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -12,6 +12,9 @@
 		25751F532372C9460019F657 /* Dcrlibwallet.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 25751F522372C9460019F657 /* Dcrlibwallet.framework */; };
 		25A4331C2373D0330018FFCF /* SyncManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = 25A4331B2373D0330018FFCF /* SyncManager.swift */; };
 		25E63F4E23312B2C009CD53E /* NavMenuFloatingButtons.swift in Sources */ = {isa = PBXBuildFile; fileRef = 25E63F4D23312B2B009CD53E /* NavMenuFloatingButtons.swift */; };
+		2506406E2304A2DE0028A9AC /* Dcrlibwallet.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 2506406D2304A2DE0028A9AC /* Dcrlibwallet.framework */; };
+		AD24C23D2343E445009583C9 /* NotificationsManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = AD24C23C2343E445009583C9 /* NotificationsManager.swift */; };
+		AD2F20CB2354AABB0084371C /* Dcrlibwallet.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = AD2F20CA2354AABB0084371C /* Dcrlibwallet.framework */; };
```
Comment (itswisdomagain) [inline Decred Wallet.xcodeproj/project.pbxproj]: Multiple references to `Dcrlibwallet.framework` added here, needs cleanup.

File: Decred Wallet.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -1062,6 +1072,7 @@
 				FRAMEWORK_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/libs",
+					"$(PROJECT_DIR)",
```
Comment (itswisdomagain) [inline Decred Wallet.xcodeproj/project.pbxproj]: This shouldn't be.

File: Decred Wallet/Features/Overview/SyncProgressViewController.swift [hunk 8]
```diff
@@ -152,6 +152,8 @@ extension SyncProgressViewController: SyncProgressListenerProtocol {
         
         // Display "show details" button if details are not being shown currently.
         self.showDetailedSyncReportButton.isHidden = !self.currentSyncActionReportLabel.isHidden
+        // Update sync notification
+
```
Comment (itswisdomagain) [inline Decred Wallet/Features/Overview/SyncProgressViewController.swift]: uhm, what's this do?

File: Decred Wallet/Utils/NotificationsManager.swift [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+            // We can update the content in the notification center when the screen is locked
+            DispatchQueue.main.async {
+                guard UIApplication.shared.isProtectedDataAvailable == false else {return}
+                self.postBackGroundSyncNotificationRequest(message: message)
+            }
+        }
+    }
+
+    func removeSyncInprogressNotification() {
```
Comment (itswisdomagain) [inline Decred Wallet/Utils/NotificationsManager.swift]: `removeSyncProgressNotification`

File: Decred Wallet/Utils/NotificationsManager.swift [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        notificationCenter.removeDeliveredNotifications(withIdentifiers: [NotificationCategory.syncInProgressNotification.rawValue])
+    }
+    
+    fileprivate func postBackGroundSyncNotificationRequest(message: String) {
```
Comment (itswisdomagain) [inline Decred Wallet/Utils/NotificationsManager.swift]: `postBackgroundSyncNotificationRequest`

File: Decred Wallet/Features/Wallet Utils/Syncer.swift [hunk 11]
```diff
@@ -237,13 +240,23 @@ class Syncer: NSObject, AppLifeCycleDelegate {
...
         if backgroundTask != .invalid {
             UIApplication.shared.isNetworkActivityIndicatorVisible = false
             print("Background task ended at: ", Date())
             UIApplication.shared.endBackgroundTask(backgroundTask)
             backgroundTask = .invalid
         }
     }
+
+    private func fireLocalBackgroundSyncNotificationIfInBackground(with progress: DcrlibwalletGeneralSyncProgress) {
+        let message = String(format: LocalizedStrings.syncTotalProgress, progress.totalSyncProgress, progress.totalTimeRemaining)
```
Comment (itswisdomagain) [inline Decred Wallet/Features/Wallet Utils/Syncer.swift]: No point creating this `message` variable if it won't be posted. I'd rather:
```
if self.syncCompletedCanceledOrErrored || UIApplication.shared.applicationState != .background {
        return
}

DispatchQueue.main.async {
        let message = String(format: LocalizedStrings.syncTotalProgress, progress.totalSyncProgress, progress.totalTimeRemaining)
        NotificationsManager.shared.fireSyncInProgressNotification(with: message)
}
```

**Pull request merged**